src/classes/Mfa.php

```diff
@@ -0,0 +1,178 @@
+<?php
+/**
+ * @package   Elabftw\Elabftw
+ * @author    Nicolas CARPi <nico-git@deltablot.email>
+ * @copyright 2012 Nicolas CARPi
+ * @license   https://www.gnu.org/licenses/agpl-3.0.html AGPL-3.0
+ * @see       https://www.elabftw.net Official website
+ */
+declare(strict_types=1);
+
+namespace Elabftw\Elabftw;
+
+use Elabftw\Services\Filter;
+use Elabftw\Services\MpdfQrProvider;
+use RobThree\Auth\TwoFactorAuth;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Session\Session;
+use function time;
+
+/**
+ * Provide methods for multi/two-factor authentication
+ */
+class Mfa
+{
+    /** @var Session $Session Current session */
+    public $Session;
+
+    /** @var Db $Db SQL Database */
+    protected $Db;
+
+    /** @var TwoFactorAuth $TwoFactorAuth PHP Class for handling two/multi-factor authentication */
+    private $TwoFactorAuth;
+
+    /** @var Request $Request Current request */
+    private $Request;
+
+    /**
+     * Constructor
+     *
+     * @param Request $request
+     * @param Session $session
+     */
+    public function __construct(Request $request, Session $session)
+    {
+        $this->TwoFactorAuth = new TwoFactorAuth('eLabFTW', 6, 30, 'sha1', new MpdfQrProvider());
+        $this->Db = Db::getConnection();
+        $this->Request = $request;
+        $this->Session = $session;
+    }
+
+    /**
+     * Test if user has 2FA activated
+     * Redirect to multi-factor code submission if active
+     *
+     * @param string $redirect Where user will be redirected to after code submission
+     * @return void
+     */
+    public function needVerification(int $userid, string $redirect): void
+    {
+        $MFASecret = $this->getSecret($userid);
+        if ($MFASecret && !$this->Session->has('mfa_verified')) {
+            $this->Session->set('mfa_secret', $MFASecret);
+            $this->Session->set('mfa_redirect', $redirect);
+
+            $Response = new RedirectResponse('../../mfa.php');
+            $Response->send();
+            exit();
+        }
+    }
+
+    /**
+     * Load MFA secret of user from databse if exists
+     *
+     * @param int $userid
+     * @return mixed MFA secret or false
```

Comment:
maybe throw exception instead of returning false. We should avoid mixed returns

Reply:
Now it returns the secret or an empty string. No more mixed return here.